ethereum/eip712/signed_data.go

```diff
@@ -0,0 +1,1053 @@
+// 	return req, useEthereumV, nil
+// }
+
+// SignTextWithValidator signs the given message which can be further recovered
+// with the given validator.
+// hash = keccak256("\x19\x00"${address}${data}).
+func SignTextValidator(validatorData ValidatorData) (hexutil.Bytes, string) {
+	msg := fmt.Sprintf("\x19\x00%s%s", string(validatorData.Address.Bytes()), string(validatorData.Message))
+	return crypto.Keccak256([]byte(msg)), msg
+}
+
+// cliqueHeaderHashAndRlp returns the hash which is used as input for the proof-of-authority
+// signing. It is the hash of the entire header apart from the 65 byte signature
+// contained at the end of the extra data.
+//
+// The method requires the extra data to be at least 65 bytes -- the original implementation
+// in clique.go panics if this is the case, thus it's been reimplemented here to avoid the panic
+// and simply return an error instead
+func cliqueHeaderHashAndRlp(header *types.Header) (hash, rlp []byte, err error) {
+	if len(header.Extra) < 65 {
+		err = fmt.Errorf("clique header extradata too short, %d < 65", len(header.Extra))
+		return
+	}
+	rlp = clique.CliqueRLP(header)
+	hash = clique.SealHash(header).Bytes()
+	return hash, rlp, err
+}
+
+// // SignTypedData signs EIP-712 conformant typed data
+// // hash = keccak256("\x19${byteVersion}${domainSeparator}${hashStruct(message)}")
+// // It returns
+// // - the signature,
+// // - and/or any error
+// func (api *SignerAPI) SignTypedData(ctx context.Context, addr common.MixedcaseAddress, typedData TypedData) (hexutil.Bytes, error) {
+// 	signature, _, err := api.signTypedData(ctx, addr, typedData, nil)
+// 	return signature, err
+// }
+
+// // signTypedData is identical to the capitalized version, except that it also returns the hash (preimage)
+// // - the signature preimage (hash)
+// func (api *SignerAPI) signTypedData(ctx context.Context, addr common.MixedcaseAddress,
+// 	typedData TypedData, validationMessages *apitypes.ValidationMessages) (hexutil.Bytes, hexutil.Bytes, error) {
+// 	domainSeparator, err := typedData.HashStruct("EIP712Domain", typedData.Domain.Map())
+// 	if err != nil {
+// 		return nil, nil, err
+// 	}
+// 	typedDataHash, err := typedData.HashStruct(typedData.PrimaryType, typedData.Message)
+// 	if err != nil {
+// 		return nil, nil, err
+// 	}
+// 	rawData := []byte(fmt.Sprintf("\x19\x01%s%s", string(domainSeparator), string(typedDataHash)))
+// 	sighash := crypto.Keccak256(rawData)
+// 	messages, err := typedData.Format()
+// 	if err != nil {
+// 		return nil, nil, err
+// 	}
+// 	req := &SignDataRequest{
+// 		ContentType: DataTyped.Mime,
+// 		Rawdata:     rawData,
+// 		Messages:    messages,
+// 		Hash:        sighash,
+// 		Address:     addr}
+// 	if validationMessages != nil {
+// 		req.Callinfo = validationMessages.Messages
+// 	}
+// 	signature, err := api.sign(req, true)
+// 	if err != nil {
+// 		api.UI.ShowError(err.Error())
+// 		return nil, nil, err
+// 	}
+// 	return signature, sighash, nil
+// }
```

Comment:
@crypto-facs I left this code and the one on top commented in case it's useful to make some tests. 
If you don't need it we can remove it.